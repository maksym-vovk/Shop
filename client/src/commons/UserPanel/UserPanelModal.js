import React, {useEffect, useRef} from 'react';
import { Link } from 'react-router-dom';

import './index.scss';

export const UserPanelModal = props => {
  const useOutsideClick = (ref) => {
    const clickOutside = (event) => {
      if (ref.current && !ref.current.contains(event.target)) {
        props.modal(false);
      }
    }
    useEffect(() => {
      document.addEventListener('mousedown', clickOutside)
      return () => {
        document.removeEventListener('mousedown', clickOutside)
      }
    })
  }

  const wrapperRef = useRef(null)
  useOutsideClick(wrapperRef)
  return (
    <div className="user-modal" ref={wrapperRef}>
      <div className="user-modal__name-block">
        <p className="user-modal__name-text">{props.name}</p>
        <p className="user-modal__name-text">{props.surname}</p>
      </div>
      <div className="user-modal__link-block">
        <Link to="/order_history" className="user-modal__link user-modal__link--margin" onClick={() => props.modal(false)}>Orders history</Link>
        <Link to="/user_cabinet" className="user-modal__link" onClick={() => props.modal(false)}>Account</Link>
      </div>
      <div className="user-modal__logout-block">
        <button className="user-modal__logout" onClick={props.logout}>Sign out</button>
      </div>
    </div>
  )
}