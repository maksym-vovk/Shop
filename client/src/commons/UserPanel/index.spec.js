import React from 'react';
import { UserPanelModal } from './UserPanelModal';

describe('UserPanelModal component', () => {
  it('UserPanelModal component render', () => {
    const wrapper = shallow(<UserPanelModal />);
    expect(wrapper).toMatchSnapshot();
  });
});