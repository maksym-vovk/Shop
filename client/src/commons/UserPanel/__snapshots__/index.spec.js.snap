// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPanelModal component UserPanelModal component render 1`] = `ShallowWrapper {}`;
